### DISCUSSION TITLE
Is there a way to run the type generation step manually?

### DISCUSSION THREAD

**Discussion by chamatt** [0no-co/gql.tada#56] [Q&A] [answered]
Hi everyone, I'm very interested in integrating this library onto the codebase I work on. Only thing I'm left wondering is if there's a way to run the introspection generation without running vscode. That would be helpful for team members that use other IDE's, as I could set it up to run on a pre-commit hook or something like that

**Comment by JoviDeCroock** [comment, marked as answer]
Any IDE (apart from jetbrains) supports tsserver plugins this isn't just for VSCode. We currently don't support standalone introspection generation, however we might in the future.

If you need it immediately this is how the lsp does it https://github.com/0no-co/GraphQLSP/blob/main/packages/graphqlsp/src/graphql/getSchema.ts

**Reply by cdrx** [reply]
OT, but:

> Any IDE (apart from jetbrains) supports tsserver plugins

I've been using this successfully with WebStorm, so I think tsserver plugins must be supported.

**Reply by JoviDeCroock** [reply]
Oh, odd, then some JetBrains IDE's do not support it 😅 was talking with someone earlier that it didn't work for PHPStorm

**Reply by donferi** [reply]
Would love to have a way to run a command and generate it. ❤️

I can't make any assumptions on the environment of my consumers. Would be awesome if I could make sure it generates for everyone. (bootstrap cli context)

**Reply by kjr-lh** [reply]
@cdrx 

Hi, did you get suggestions / autocomplete for fields working in webstorm? I know the plugin is being used in my project as I get type errors in the ide if I try to use invalid fields, but I'm really missing suggestions...

**Reply by yuthikaxsagarage** [reply]
i cannot get autocomplete to work with webstorm, i just gave it a go for experimentation. Same here missing the suggestions, on the data.
edit- after manual introspection it works on webstorm
